### PR TITLE
DAT-23: BE - Mail to confirm

### PR DESCRIPTION
**Pull request by thanhphuchuynh**
Firstly, change email account and password conf in config XML.
image

API get: _`/emails?email={email}`_ 

image

Result:  `code expired: 120s`
image

API get: _`/confirmation?email={email}&code={code}`_ 
ex: {{local }}/confirmation?email=nguoigiaumat100@gmail.com&code=NeCB59

image

UT

image

### CODE REVIEW

File: internal/app/api/repositories/user/user_repo.go [hunk 1]
```diff
@@ -310,6 +310,23 @@ func (r *MongoRepository) CountUserUsersAvailable(ctx context.Context, ignoreIds
 	return r.collection().CountDocuments(ctx, query)
 }
 
+// this method help get update verify email in mail collection
+func (r *MongoRepository) UpdateMailVerified(ctx context.Context, email string) error {
```
Comment (paphuc) [inline internal/app/api/repositories/user/user_repo.go]: Can we move this to mail repo?

File: internal/app/api/services/mail/mail.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	//send email
+	if err := mailpkg.Sendmail(mailpkg.Mail{
+		Subject: "Mail Confirm",
+		Body:    code,
+	}, []string{mail}, s.conf); err != nil {
+		return errors.Wrap(err, "Send mail fail")
+	}
+
+	//upsert db
+	if err := s.repo.Insert(ctx, types.EmailVerification{
```
Comment (paphuc) [inline internal/app/api/services/mail/mail.go:106]: Please save code to DB before sending mail

**Comment by paphuc** [comment]
Please update the unit test

File: internal/app/api/repositories/user/user_repo.go (hunk 1 above)
Comment (thanhphuchuynh) [reply internal/app/api/repositories/user/user_repo.go]: Mail Verified was updated after the user successful signup, so I think it should be in user repo...
If we move this to mail repo, maybe create 1 API  update, I think this not good

**Pull request merged**